[typescript] Unable to unsubscribe from WebRequest events

* **Electron Version** (output of `node_modules/.bin/electron --version`):
  * 3.10.0
* **Operating System** (Platform and Version):
  * macOS 10.12.6
* **Last known working Electron version** (if applicable):
  * None

### Expected Behavior
When using Electron with TypeScript, I can pass `null` to the `WebRequest.onBeforeSendHeaders()` function in order to remove the listener as described here:

https://electronjs.org/docs/api/web-request

### Actual behavior
The defined TypeScript types do not allow passing null to this function nor the others in this class.

### To Reproduce
Try compiling TypeScript such as:

```typescript
session.webRequest.onBeforeSendHeaders(null);
```

tsc throws an error:

```bash
error TS2345: Argument of type 'null' is not assignable to parameter of type 'Function'.
```

I was unable to see how types are generated, so a PR isn't forthcoming.

--- a/docs/api/web-request.md
+++ b/docs/api/web-request.md
@@ -45,7 +45,7 @@ The following methods are available on instances of `WebRequest`:
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -72,7 +72,7 @@ The `callback` has to be called with an `response` object.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -98,7 +98,7 @@ The `callback` has to be called with an `response` object.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -117,7 +117,7 @@ response are visible by the time this listener is fired.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -147,7 +147,7 @@ The `callback` has to be called with an `response` object.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -170,7 +170,7 @@ and response headers are available.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -193,7 +193,7 @@ redirect is about to occur.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String
@@ -215,7 +215,7 @@ completed.
 * `filter` Object (optional)
   * `urls` String[] - Array of URL patterns that will be used to filter out the
         requests that do not match the URL patterns.
-* `listener` Function
+* `listener` Function | null
   * `details` Object
     * `id` Integer
     * `url` String

